### PR TITLE
ddns-scripts: Add script for TransIP.nl

### PR DESCRIPTION
**Pull request by Atema**
Maintainer: @chris5560 / @feckert
Compile tested: No
Run tested: Manual installation of the script on openwrt 21.02 with the current version of ddns-scripts, curl, and openssl-utils installed works as expected.

Description: Adds a custom script for updating DNS records at TransIP.nl (one of the largest registrars in The Netherlands) with ddns-scripts

### CODE REVIEW

File: net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+[ -z "${CURL_SSL}" ] && write_log 14 "TransIP update requires cURL with SSL"
+[ -z "$(openssl version)" ] && write_log 14 "TransIP update requires openssl binary"
+
+. /usr/share/libubox/jshn.sh
+
+
+# Re-format the private key and write to a temporary file
+
+__tmp_keyfile="/tmp/ddns-transip-$(openssl rand -hex 4)"
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh:39]: Please use mktemp for this
https://github.com/openwrt/packages/search?q=mktemp

File: net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        -H "Content-Type: application/json" \
+        -H "Signature: ${__auth_signature}" \
+        -d "${__auth_body}" \
+        -w "%{http_code}\n" \
+        -o "${DATFILE}" 2>"${ERRFILE}")
+
+
+# Logging for error and debug
+
+if [ $? -ne 0 ]
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh:81]: Please use `if [ $? -ne 0 ]; then`

File: net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+then
+        write_log 14 "Curl failed: $(cat "${ERRFILE}")"
+        return 1
+fi
+
+if [ -z ${__auth_status} ] || [ ${__auth_status} -ne 201 ]
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh:87]: Please use ` ]; then`

File: net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        -H "Content-Type: application/json" \
+        -H "Authorization: Bearer ${__auth_token}" \
+        -d "${__update_body}" \
+        -w "%{http_code}\n" \
+        -o "${DATFILE}" 2>"${ERRFILE}")
+
+
+# Logging for error and debug
+
+if [ $? -ne 0 ]
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh:125]: Please use ` ]; then`

File: net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+then
+        write_log 14 "Curl failed: $(cat "${ERRFILE}")"
+        return 1
+fi
+
+if [ -z ${__update_status} ] || [ ${__update_status} -ne 204 ]
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_transip_nl.sh:131]: Please use ` ]; then`

**Review by feckert — CHANGES_REQUESTED** [review]
Unfortunately, I can't test this because I don't have an account with this provider. Everything else looks good so far

**Comment by Atema** [comment]
@feckert I've updated with the requested changes. Can I assume that `mktemp` is always available?

**Comment by feckert** [comment]
In the openwrt repository the requested busybox app is set to `y`
https://github.com/openwrt/openwrt/blob/b6cbbbb6ef4deaba00f5c128ea66696fc64c5bf0/package/utils/busybox/Config-defaults.in#L703
I don't expect it to change, but to be on the safe side we could add this as a dependency to.
https://github.com/openwrt/packages/pull/17136/files#diff-c0e9e626f2ba612f4284522051479b373f8481892313fdc5c766dae5e70cbf1dR213

Have a look how this could be done in openwrt https://github.com/openwrt/openwrt/blob/master/package/network/utils/wireguard-tools/Makefile#L40

**Comment by Atema** [comment]
@feckert I added a dependency for `coreutils-mktemp` only if the symbol `BUSYBOX_CONFIG_MKTEMP` is not set (using syntax from https://openwrt.org/docs/guide-developer/dependencies). That should make it work in any scenario.

**Comment by feckert** [comment]
@Atema thanks for this :+1: . I have update the PKG_RELEASE number so I pushed it directly.
https://github.com/openwrt/packages/commit/aa820a2cae321311097e59de9b67dfa5c648cbeb

**Pull request closed without merging**